### DISCUSSION TITLE
Selectable Text in Label

### DISCUSSION THREAD

**Discussion by maixnor** [emilk/egui#2216] [Q&A]
So I've made an image to ascii converter and I would like the result to be selectable to copy it.

So far I could not find any documentation on this, but it has to be possible since there is a portion on www.egui.rs that does exactly what I want (Demos>About EGUI>Button snippet).

so far I am using `ui.monospace(text)` but that does not do the trick.

You can check it out [here](ascii.maixnor.com) with source here [maixnor/ascii](github.com/maixnor/ascii).

Thank you for your time, already in advance!!!!!!

**Comment by maixnor** [comment]
This is the relevant snipped at `src/app.rs:152`

```
fn display_image_bytes(ui: &mut egui::Ui, bytes: &[u8]) {
    use std::io::Cursor;

    let reader = image::io::Reader::new(Cursor::new(bytes))
        .with_guessed_format()
        .unwrap();

    match reader.decode() {
        Ok(img) => ui.monospace(convert_to_ascii(img)),
        Err(_) => ui.label("Not a valid Image!"),
    };
}
```

**Comment by ItsEthra** [comment]
Will this work for you?
```rust
ui.add(Label::new(RichText::new("your_text").monospace()).sense(Sense::click())).clicked() {
     // stuff
}
```

**Comment by elliottslaughter** [comment]
I wanted to know about this too, so I dug into the demo code. You can find the relevant code snippet here:

https://github.com/emilk/egui/blob/6ae4bc486b2608212f5fc8453f5aa7e5106853f8/crates/egui_demo_lib/src/demo/about.rs#L59-L66

This calls into a `code_view_ui` which is defined here:

https://github.com/emilk/egui/blob/6ae4bc486b2608212f5fc8453f5aa7e5106853f8/crates/egui_demo_lib/src/syntax_highlighting.rs#L4

As best I can tell, the relevant portion of this is the fact that the text is placed into a `TextEdit`:

https://github.com/emilk/egui/blob/6ae4bc486b2608212f5fc8453f5aa7e5106853f8/crates/egui_demo_lib/src/syntax_highlighting.rs#L14-L21

One thing I'm puzzling over is why the code snippet isn't editable, while the very similar looking `TextEdit` demo is. I guess it's just because the value keeps getting overwritten by the original code on every frame. So the user really is editing it, but the edits get thrown away.

For comparison, an editable `TextEdit`:

https://github.com/emilk/egui/blob/6ae4bc486b2608212f5fc8453f5aa7e5106853f8/crates/egui_demo_lib/src/demo/text_edit.rs#L44-L46

**Reply by CyanNex** [reply]
> One thing I'm puzzling over is why the code snippet isn't editable

It's because the text is a string slice. If the text is a mutable `String` type, the field will be editable. However, if you put a (mutable) `&str` in there, it won't be editable.

```rust
TextEdit::multiline(&mut "test")
     .desired_rows(1)
     .show(ui);
```
